superset/views.py

```diff
@@ -1294,6 +1294,10 @@ def clean_fulfilled_requests(session):
             for r in session.query(DAR).all():
                 datasource = SourceRegistry.get_datasource(
                     r.datasource_type, r.datasource_id, session)
+                if not datasource:
```

Comment:
if not datasource or self.datasource_access(datasource, user):
  session.delete(r)